### ISSUE TITLE
Server Demo: java.lang.IllegalArgumentException

### ISSUE THREAD

**Issue by boaks** [eclipse/leshan#116]
Writing on `/5/0/1` "Firmware Update package UR"I I get:

```
java.lang.IllegalArgumentException: The validated object is null
	at org.eclipse.leshan.util.Validate.notNull(Validate.java:222) ~[classes/:?]
	at org.eclipse.leshan.util.Validate.notNull(Validate.java:205) ~[classes/:?]
	at org.eclipse.leshan.core.request.WriteRequest.<init>(WriteRequest.java:216) ~[classes/:?]
	at org.eclipse.leshan.core.request.WriteRequest.<init>(WriteRequest.java:209) ~[classes/:?]
	at org.eclipse.leshan.server.demo.servlet.ClientServlet.writeRequest(ClientServlet.java:381) ~[classes/:?]
	at org.eclipse.leshan.server.demo.servlet.ClientServlet.doPut(ClientServlet.java:173) [classes/:?]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:710) [javax.servlet-api-3.1.0.jar:3.1.0]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790) [javax.servlet-api-3.1.0.jar:3.1.0]
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:738) [jetty-servlet-9.1.4.v20140401.jar:9.1.4.v20140401]
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:551) [jetty-servlet-9.1.4.v20140401.jar:9.1.4.v20140401]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143) [jetty-server-9.1.4.v20140401.jar:9.1.4.v20140401]
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:568) [jetty-security-9.1.4.v20140401.jar:9.1.4.v20140401]
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:221) [jetty-server-9.1.4.v20140401.jar:9.1.4.v20140401]
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1111) [jetty-server-9.1.4.v20140401.jar:9.1.4.v20140401]
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:478) [jetty-servlet-9.1.4.v20140401.jar:9.1.4.v20140401]
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:183) [jetty-server-9.1.4.v20140401.jar:9.1.4.v20140401]
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1045) [jetty-server-9.1.4.v20140401.jar:9.1.4.v20140401]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) [jetty-server-9.1.4.v20140401.jar:9.1.4.v20140401]
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97) [jetty-server-9.1.4.v20140401.jar:9.1.4.v20140401]
	at org.eclipse.jetty.server.Server.handle(Server.java:462) [jetty-server-9.1.4.v20140401.jar:9.1.4.v20140401]
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:279) [jetty-server-9.1.4.v20140401.jar:9.1.4.v20140401]
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:232) [jetty-server-9.1.4.v20140401.jar:9.1.4.v20140401]
	at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:534) [jetty-io-9.1.4.v20140401.jar:9.1.4.v20140401]
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:607) [jetty-util-9.1.4.v20140401.jar:9.1.4.v20140401]
	at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:536) [jetty-util-9.1.4.v20140401.jar:9.1.4.v20140401]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_51]
```

Seems to be caused by:

```
    private WriteRequest(final Mode mode, ContentFormat format, final LwM2mPath target, final LwM2mNode node) {
    ...
        Validate.notNull(format);

```

called in ClientServlet

```
private WriteResponse writeRequest(Client client, String target, HttpServletRequest req, HttpServletResponse resp)
            throws IOException, InterruptedException {
            ...
            return server.send(client, new WriteRequest(Mode.REPLACE, null, target, node), TIMEOUT);


```

**Comment by sbernard31** [comment]
This is a regression, I introduced because I want to respect this part of the spec :

> "The request includes the value to be written in the corresponding Plain Text, Opaque, TLV or JSON format according to the Content-Format option which MUST be specified [CoAP]."

I forgot that currently we do some magic to guess the content-format at coap request creation.. So the `Validate.notNull(format)` is not necessary. I will remove it.
Later we will use the TLV format by default if `format` is null as the specification allow it for all LWM2M node now.  We will be able to remove this magic !

Thx @boaks

**Comment by sbernard31** [comment]
It should be fixed in master (since  commit a4db6164d022064b02cefc6346fa786abc4303e3)
